### ISSUE TITLE
Create/Assign @type to info.json resources

### ISSUE THREAD

**Issue by azaroth42** [IIIF/api#1121]
The proposal is to add a *mandatory* `@type` to info.json resources, such as `iiif:ImageService` in 3.0
The rationale is a bit longer to explain...

In JSON-LD 1.1 there will be the facility to do per-predicate and per-class (`@type`) contexts, nested within the main context document.  This is described here: http://json-ld.org/spec/latest/json-ld/#scoped-contexts

We can avoid the current known issue of having to use `@context` keys within the structure by having sufficient hooks to refer to them by. When expanding the JSON-LD to RDF, the contexts disappear as they're not themselves triples. Thus when expanding (e.g. as part of framing) and then later compacting back to the recommended JSON form, the contexts are lost.  With scoped contexts, the main context keeps the references rather than the data.  Non-core extensions would still need to be re-injected ... but we have yet to see [m]any of these.

As our extension point is just `service`, we can't rely on predicate based scoped contexts, so need to use class based, and our core services currently don't have them.

In (e.g.) the Presentation context we would have:

```
  "IIIFImageService": {"@id": "iiif:ImageService", 
      "@type": "@id", 
      "@context": "http://iiif.io/api/image/3/context.json"},
```

And then in the data:

```
  "service": {
      "id": "http://example.org/iiif/identifier",
      "type": "IIIFImageService",
      "profile": "http://iiif.io/api/image/2/level1.json"
  }
```

And then the same for all other services that we intend to embed, notably auth and search.

**Comment by azaroth42** [comment]
Impacts #1012

**Comment by azaroth42** [comment]
See #1185

**Comment by azaroth42** [comment]
Proposal:  Services will have types, these would be specific to major versions e.g. "ImageService3" so they can be tied to specific contexts. Other services SHOULD use "Service", but it would not be non-conforming to have no `type`.

**Comment by azaroth42** [comment]
Note that the `type` replaces `@context` on the service's description block and thus needs the major version, but does _not_ replace `profile`, which encodes the available functionality.

There should (following #1284, if accepted) be aliases in the context for the definition of profile values.

Thus the above example would become:
```
{
  "id": "http://example.net/iiif/image/1",
  "type": "ImageService3",
  "profile": "level1"
}
```

**Comment by zimeon** [comment]
:+1: agreed at Toronto WG meeting

Some discussion of where `ImageService2` might be defined. Agreement that this doesn't need ot go in the old specs but might go in an Annex as well as in the new context.

**Comment by azaroth42** [comment]
Eds discussion 12/13:  Put a callout box in Prezi3 for referring to other IIIF types, and define ImageService2 as a special case there. Doesn't need to be refered to from other existing specs, and future specs are less likely to have the problem.

**Comment by azaroth42** [comment]
(And ImageService1)

**Comment by zimeon** [comment]
See #1430 and #1431 for corresponding issues for next versions of auth and search

**Comment by azaroth42** [comment]
Closed by #1429
